tests/bud.bats

```diff
@@ -117,6 +117,61 @@ symlink(subdir)"
   expect_output --substring $(realpath "${TESTSDIR}/bud/dockerignore3/.dockerignore")
 }
 
+# Following test must fail since we are trying to run linux/arm64 on linux/amd64
+# Issue: https://github.com/containers/buildah/issues/3712
+@test "build-with-inline-platform" {
+  # Host arch
+  run_buildah info --format '{{.host.arch}}'
+  myarch="$output"
+  otherarch="arm64"
+
+  # just make sure that other arch is not equivalent to host arch
+  if [[ "$otherarch" == "$myarch" ]]; then
+    otherarch="amd64"
+  fi
+  # ...create a Containerfile with --platform=linux/$otherarch
+  cat > ${TESTSDIR}/bud/platform/Dockerfile << _EOF
+FROM --platform=linux/${otherarch} alpine
+RUN uname -m
+_EOF
+
+  run_buildah '?' build --signature-policy ${TESTSDIR}/policy.json -t test ${TESTSDIR}/bud/platform
+  if [[ $status -eq 0 ]]; then
+    # Build succeeded: we must have qemu-user-static installed. Confirm that
+    # the 'uname' emits what we expect
+    expect_output --from="${lines[-1]}" "$otherarch"
+    run_buildah inspect --format '{{ .OCIv1.Architecture }}' test
+    expect_output --substring "$otherarch"
+  else
+    # Build failed: we DO NOT have qemu-user-static installed.
+    expect_output --substring "format error"
+  fi
+}
+
+# Following test must pass since we want to tag image as host arch
+# Test for use-case described here: https://github.com/containers/buildah/issues/3261
+@test "build-with-inline-platform-amd-but-tag-as-arm" {
+  # Host arch
+  run_buildah info --format '{{.host.arch}}'
+  myarch="$output"
```

Comment:
Might be simpler to grep the output of `buildah version`'s `BuildPlatform`, which provides the entire value you're looking to use.

Reply:
@nalind I need to toggle  only `arch` at certain places that's why i preferred this. Hope this is not a blocker.